app/scripts/proactive/config.js

```diff
@@ -273,91 +273,128 @@ println "DOCKER COMMAND : " + forkEnvironment.getPreJavaCommand()
 // Prepare Singularity parameters
 import org.ow2.proactive.utils.OperatingSystem;
 import org.ow2.proactive.utils.OperatingSystemFamily;
+import org.codehaus.groovy.runtime.StackTraceUtils;
 
-String osName = System.getProperty("os.name");
-println "Operating system : " + osName;
-OperatingSystem operatingSystem = OperatingSystem.resolveOrError(osName);
-OperatingSystemFamily family = operatingSystem.getFamily();
+imageUrl = "docker://java"
+imageName = imageUrl.substring(imageUrl.indexOf("://") + 3).replace("/","_").replace(":","_")
+imageLockFileName = imageName + ".lock"
+userHome = System.getProperty("user.home")
 
-switch (family) {
-    case OperatingSystemFamily.WINDOWS:
+try {
+
+    String osName = System.getProperty("os.name");
+    println "Operating system : " + osName;
+    OperatingSystem operatingSystem = OperatingSystem.resolveOrError(osName);
+    OperatingSystemFamily family = operatingSystem.getFamily();
+
+    switch (family) {
+        case OperatingSystemFamily.WINDOWS:
         throw new IllegalStateException("Singularity is not supported on Windows operating system")
-    case OperatingSystemFamily.MAC:
-    	throw new IllegalStateException("Singularity is not supported on Mac operating system")
-    default:
-        isWindows = false;
-}
+        case OperatingSystemFamily.MAC:
+        throw new IllegalStateException("Singularity is not supported on Mac operating system")
+        default:
+            isWindows = false;
+    }
 
-containerName = "java"
-containerUrl = "docker://" + containerName
+    process = "singularity --version".execute()
+    majorVersion = 0
+    if (process.waitFor() == 0) {
+        version = process.text
```

Comment:
we can display singularity version by adding:
println process.text OR println version

Reply:
right, just added it